ignite/metrics/accuracy.py

```diff
@@ -18,16 +18,17 @@ def reset(self):
         self._type = None
         self._num_classes = None
 
+    def _process_shape(self, y):
```

Comment:
This method is ugly, sorry

Reply:
I would prefer not to squeeze at all and adapt our logic and raise an error if we can not understand the case.

Reply:
I updated the function. 

What logic would you use to check the shapes?

Reply:
I'm trying to analyse here. 
As documentation states:
```
    - `y_pred` must be in the following shape (batch_size, num_categories, ...) or (batch_size, ...).
    - `y` must be in the following shape (batch_size, ...).
```
we must have batch size of `y` and batch size of `y_pred` to be equal and all dimensions hided behind `...` to be equal for `y` and `y_pred`.

This logic should be sufficient ?

Next, we need to determine the case : multi-class or (binary/multi-label). I think `y.ndimension() + 1 == y_pred.ndimension()` should hold even if we do not squeeze 1's.

Reply:
@vfdev-5 I mostly agree. Here's a situation.

Let's take an example of a binary case of batch size N - model output shape is (N, 1) and training set shape is (N, ). Given logic of check_type, this case will be considered a multiclass case, because y_pred.ndimension() == y.ndimension() + 1.

Should we have users account for situations using output_transform? I personally think we should account for this situation in the metrics. Most updated way of handling 1's in the shape, I think works well.

Reply:
Or we can assign a multi-class case when second dimension is larger than one ?

Reply:
Clarifying, if a user inputs (N, 10) and (N, 1) for multiclass cases. That is considered wrong by the new logic right? We should raise an error for this?

Reply:
We can follow this decision "tree" (https://github.com/pytorch/ignite/pull/275#issuecomment-425396475) and update it to the current state.